Signal/src/ViewControllers/ConversationView/ConversationViewModel.m

```diff
@@ -863,6 +900,25 @@ - (BOOL)reloadViewItems
             OWSAssertDebug(!viewItemCache[interaction.uniqueId]);
             viewItemCache[interaction.uniqueId] = viewItem;
         }
+
+        if (self.typingIndicatorsRecipient) {
+            id<ConversationViewItem> _Nullable lastViewItem = viewItems.lastObject;
+            uint64_t typingIndicatorTimestamp = (lastViewItem ? lastViewItem.interaction.timestamp + 1 : 1);
```

Comment:
The typing indicator use a fake timestamp after the last interaction in the conversation - or 1 if none exists.